packages/openapi-v3/src/extension-point/spec-contributor.service.ts

```diff
@@ -0,0 +1,69 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/example-greeter-extension
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {config, Getter} from '@loopback/context';
+import {extensionPoint, extensions} from '@loopback/core';
+import * as debugModule from 'debug';
+import * as _ from 'lodash';
+import {inspect} from 'util';
+import {OpenApiSpec} from '../types';
+import {OAISpecEnhancer, OAISPEC_ENHANCER_EXTENSION_POINT_NAME} from './types';
+const debug = debugModule('loopback:openapi:spec-contributor');
+
+/**
+ * Options for the OAI Spec contributor extension point
+ */
+export interface OAISpecContributorServiceOptions {
+  // no-op
+}
+
+/**
+ * An extension point for OAI contributors
+ */
+@extensionPoint(OAISPEC_ENHANCER_EXTENSION_POINT_NAME)
+export class OAISpecEnhancerService {
+  constructor(
+    /**
+     * Inject a getter function to fetch spec contributors
+     */
+    @extensions()
+    private getContributors: Getter<OAISpecEnhancer[]>,
+    /**
+     * An extension point should be able to receive its options via dependency
+     * injection.
+     */
+    @config()
+    public readonly options?: OAISpecContributorServiceOptions,
+  ) {}
+
+  private _spec: OpenApiSpec = {
+    openapi: '3.0.0',
+    info: {
+      title: 'LoopBack Application',
+      version: '1.0.0',
+    },
+    paths: {},
+  };
+
+  get spec(): OpenApiSpec {
+    return this._spec;
+  }
+  set spec(value: OpenApiSpec) {
```

Comment:
Copied from chat with @raymondfeng 

> when openapi spec is generated, we first build it from controller inspections, then call enhancers to update the spec

I add a setter for the `_spec` here for ^